zerver/views/message_flags.py

```diff
@@ -67,19 +64,20 @@ def update_message_flags(
 
 # NOTE: If this function name is changed, add the new name to the
 # query in get_latest_update_message_flag_activity
-@has_request_variables
+@typed_endpoint
 def update_message_flags_for_narrow(
     request: HttpRequest,
     user_profile: UserProfile,
-    anchor_val: str = REQ("anchor"),
-    include_anchor: bool = REQ(json_validator=check_bool, default=True),
-    num_before: int = REQ(converter=to_non_negative_int),
-    num_after: int = REQ(converter=to_non_negative_int),
-    narrow: OptionalNarrowListT = REQ("narrow", converter=narrow_parameter),
-    operation: str = REQ("op"),
-    flag: str = REQ(),
+    *,
+    anchor: str,
+    include_anchor: Json[bool] = True,
+    num_before: Json[NonNegativeInt],
+    num_after: Json[NonNegativeInt],
+    narrow: Json[OptionalNarrowListT],
```

Comment:
I think more validation is required; `narrow_parameter` did a lot more than just checking that it's an `OptionalNarrowListT`